checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -319,35 +319,34 @@ public boolean derivedFromMustCallAlias() {
 
     /**
      * Gets the must-call methods (i.e. the list of methods that must be called to satisfy the
-     * must-call obligation) of the resource represented by this Obligation.
+     * must-call obligation) of each resource alias represented by this Obligation.
      *
      * @param rlAtf a Resource Leak Annotated Type Factory
      * @param mcStore a CFStore produced by the MustCall checker's dataflow analysis. If this is
      *     null, then the default MustCall type of each variable's class will be used.
-     * @return the list of must-call method names, or null if the resource's must-call obligations
-     *     are unsatisfiable (i.e. its value in the Must Call store is MustCallUnknown)
+     * @return a map from each resource alias of this to a list of its must-call method names, or
+     *     null if the must-call obligations are unsatisfiable (i.e. the value of some tracked
+     *     resource alias of this in the Must Call store is MustCallUnknown)
      */
-    public @Nullable List<String> getMustCallMethods(
+    public @Nullable Map<ResourceAlias, List<String>> getMustCallMethods(
         ResourceLeakAnnotatedTypeFactory rlAtf, @Nullable CFStore mcStore) {
+      Map<ResourceAlias, List<String>> result = new HashMap<>(this.resourceAliases.size());
       MustCallAnnotatedTypeFactory mustCallAnnotatedTypeFactory =
           rlAtf.getTypeFactoryOfSubchecker(MustCallChecker.class);
 
-      // Need to get the LUB (ie, union) of the MC values, because if a CreatesMustCallFor
-      // method was called on just one of the aliases then they all need to be treated as if
-      // they need to call the relevant methods.
```

Comment:
I _think_ this LUB computation was not necessary to handle CreatesMustCallFor even before this PR, since when handling CreatesMustCallFor, we remove the previous obligation (with the previous aliases) and create a new obligation with just the relevant alias.  Newly-created aliases should have an appropriate obligation, I think.  At the least, all the tests pass without this.

Reply:
This change makes me very nervous, but I'm glad that all the tests are passing.

I went digging through the git blames, and this LUB has been in the logic here since [this ancient commit](https://github.com/kelloggm/object-construction-checker/commit/ce5427f29e49bae4072b74cf290432fd3a8d8d57#diff-d9c49afa35f8d4df342e326e80fd73fd3d5df0b9c0d3a1c95cd6662f84251d00), which doesn't appear to have a corresponding test case. So, I wrote one:
```
// Test case for taking the least upper bound of obligations.

import org.checkerframework.checker.mustcall.qual.*;

public class CreatesMustCallForTwoAliases {
  @InheritableMustCall("a")
  static class Foo {

    @SuppressWarnings("mustcall")
    @MustCall() Foo() {
      // unconnected socket like
    }

    @CreatesMustCallFor("this")
    void reset() {}

    void a() {}
  }

  public static void test1() {
    Foo a = new Foo();
    // :: error: required.method.not.called
    Foo b = a;
    b.reset();
  }

  @CreatesMustCallFor("#1")
  public static void sneakyReset(Foo f) {
    f.reset();
  }

  public static void test2() {
    Foo a = new Foo();
    // :: error: required.method.not.called
    Foo b = a;
    sneakyReset(b);
  }

  public static void test3(Foo b) {
    Foo a = new Foo();
    // :: error: required.method.not.called
    b = a;
    sneakyReset(b);
  }

  public static void test4(Foo b) {
    // :: error: required.method.not.called
    Foo a = new Foo();
    b = a;
    sneakyReset(a);
  }
}
```

This test passes on the current `master` branch. Can you add it to this PR to double-check that this change isn't breaking?

Reply:
Thanks, Martin.  This passed for me locally.  The code here makes me less concerned about this change:

https://github.com/typetools/checker-framework/blob/365d31741fb4296b1419f1c639487aa5ee2499c4/checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java#L748-L769

Basically, if we are already tracking an obligation for an alias, and then we see a CreatesMustCallFor call passing that alias, we get rid of the previous obligation and create a new one tracking just that alias.  Assuming the must-call type system correctly tracks the obligation of that alias from that program point forward, we shouldn't erroneously assume that it has no obligation.  And the only other aliases in the new obligation will be ones created _after_ the CreatesMustCallFor call.

Reply:
Thanks. I'm convinced that removing the lub here is safe.